### PR TITLE
job-socket: coerce from and to in _jut_time_bounds to dates

### PR DESCRIPTION
**Pull request by go-oleg**
Was added to outrigger [here](https://github.com/juttle/outrigger/pull/25) but didn't make it into juttle-client-library when it was extracted.

@mnibecker

### CODE REVIEW

File: test/utils/job-socket.spec.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            done();
+        });
+        jobSocket = new JobSocket("ws://localhost:8080");
+    });
+
+    describe("converts time strings to dates", () => {
+        it("time field", (done) => {
+            let sampleDate = new Date(2000);
+
+            jobSocket.on("message", (payload) => {
```
Comment (mnibecker) [inline test/utils/job-socket.spec.js]: should this be `jobSocket.once`?

Comment (mnibecker) [inline test/utils/job-socket.spec.js]: actually we don't have it, but we should probably unsubscribe when we're done

File: test/utils/job-socket.spec.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        });
+    });
+
+    afterEach(() => {
+        mockServer.close();
+    });
+
+    after(() => {
+        global.WebSocket = _origWebSocket;
+    });
```
Comment (mnibecker) [inline test/utils/job-socket.spec.js]: Maybe place these at the top of your subscribe, just so you know they happen on every.

**Comment by mnibecker** [comment]
lgtm

File: test/utils/job-socket.spec.js (hunk 2 above)
Comment (go-oleg) [inline test/utils/job-socket.spec.js]: done.

**Pull request merged**